compiler/src/iree/compiler/Codegen/Common/GPU/GPUPromoteMatmulOperands.cpp

```diff
@@ -29,14 +29,37 @@ namespace mlir::iree_compiler {
 
 namespace {
 /// Helper to insert copy with derived thread config.
-Value promoteValue(OpBuilder &builder, Location loc, Value v) {
+Value promoteValue(OpBuilder &builder, Location loc, Value v,
+                   bool useDirectLoad) {
   auto tensorType = cast<RankedTensorType>(v.getType());
   SmallVector<OpFoldResult> mixedSizes = tensor::getMixedSizes(builder, loc, v);
+
   Value empty = builder.create<tensor::EmptyOp>(loc, mixedSizes,
                                                 tensorType.getElementType());
   auto copy = builder.create<linalg::CopyOp>(loc, v, empty);
-  setLoweringConfig(
-      copy, IREE::GPU::DerivedThreadConfigAttr::get(builder.getContext()));
+
+  /*
+  if (useDirectLoad) {
+    Attribute addressSpace = gpu::AddressSpaceAttr::get(
+        builder.getContext(), gpu::GPUDialect::getWorkgroupAddressSpace());
+    auto alloc = builder.create<bufferization::AllocTensorOp>(loc, tensorType,
+                                                              ValueRange{});
+    alloc.setMemorySpaceAttr(addressSpace);
+    auto copy = builder.create<linalg::CopyOp>(loc, v, alloc.getResult());
+    auto globalLoadDMAAttr =
+        IREE::GPU::UseGlobalLoadDMAAttr::get(builder.getContext());
+    setLoweringConfig(copy, globalLoadDMAAttr);
+    return copy.getResult(0);
+  }
+  */
+  if (useDirectLoad) {
+    auto globalLoadDMAAttr =
+        IREE::GPU::UseGlobalLoadDMAAttr::get(builder.getContext());
+    setLoweringConfig(copy, globalLoadDMAAttr);
+  } else {
+    setLoweringConfig(copy, IREE::GPU::DerivedThreadConfigAttr::get(
+                                builder.getContext()));
+  }
```

Comment:
style nit: make them consistent. I'm +1 on `setLoweringConfig(copy, ...::get(builder.getContext())` style because having a variable does not help readability to me.